File: NHSOnline.Backend.Worker.Mocking/MockingConfiguration.cs
using NHSOnline.Backend.Worker.Mocking.Emis;

namespace NHSOnline.Backend.Worker.Mocking
{
    public class MockingConfiguration
    {
        internal string WiremockBaseUrl { get; }
        internal EmisConfiguration EmisConfiguration { get; }

        public MockingConfiguration(string wiremockBaseUrl, EmisConfiguration emisConfiguration)
        {
            WiremockBaseUrl = wiremockBaseUrl;
            EmisConfiguration = emisConfiguration;
        }
    }
}

File: bddtests/src/main/kotlin/mocking/MockingConfiguration.kt
package mocking

import mocking.emis.EmisConfiguration

class MockingConfiguration {
    private val wiremockBaseUrl: String
    val emisConfiguration: EmisConfiguration
    val wiremockAdminUrl: String

    constructor(wiremockBaseUrl: String, emisConfiguration: EmisConfiguration) {
        this.emisConfiguration = emisConfiguration
        this.wiremockBaseUrl = if (wiremockBaseUrl.endsWith("/")) wiremockBaseUrl else "$wiremockBaseUrl/"
        this.wiremockAdminUrl = "${this.wiremockBaseUrl}__admin"
    }
}
